content/general/aws-authentication.textile

```diff
@@ -0,0 +1,212 @@
+
+4. Click "Edit trust relationship".
+
+<a href="/images/screenshots/iam/trust-relationships.png" target="_blank">
+  <img src="/images/screenshots/iam/trust-relationships.png" style="width: 100%" alt="Trust relationships">
+</a>
+
+5. Check that the JSON is similar to as follows:
+
+```[json]
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::203461409171:root"
+      },
+      "Action": "sts:AssumeRole",
+      "Condition": {
+        "StringEquals": {
+          "sts:ExternalId": "<Your_Ably_Account_ID>.<Your_Ably_app_ID>"
+        }
+      }
+    }
+  ]
+}
+```
+
+Note, this should have been created for you, based on the information you provided when you created the role.
+
+6. Click "Cancel".
+
+You now have a role that will be used by the Ably AWS account, but your role will not have permissions to invoke your Lambda function. You need to create a suitable policy and attach it to this role.
+
+h3(#create-policy). Create a policy
+
+To create a policy:
+
+1. In the IAM console sidebar select "Policies"
+
+<a href="/images/screenshots/iam/create-policy.png" target="_blank">
+  <img src="/images/screenshots/iam/create-policy.png" style="width: 100%" alt="Create policy">
+</a>
+
+2. Click "Create Policy".
+
+3. Choose "Lambda" as the service. (Or other AWS service as applicable, such as AWS SQS, or AWS Kinesis).
+
+<a href="/images/screenshots/iam/select-service.png" target="_blank">
+  <img src="/images/screenshots/iam/select-service.png" style="width: 80%" alt="Select service">
+</a>
+
+4. Type "Invoke" into the filter actions box.
+
+5. Select the "InvokeFunction" checkbox.
+
+<a href="/images/screenshots/iam/select-action.png" target="_blank">
+  <img src="/images/screenshots/iam/select-action.png" style="width: 80%" alt="Select action">
+</a>
+
+**Note:** Ably generally uses the batch version of the action where relevant; for example, for AWS SQS, Ably uses the @SendMessageBatch@ action, not the @SendMessage@ action.
+
+6. For "Resources" you will enter the ARN of your Lambda function as the specific resource. 
+
+7. Click "Add ARN". Enter the region your Lambda function is hosted in. Also enter the name of the Lambda function.
+
+<a href="/images/screenshots/iam/add-arn.png" target="_blank">
+  <img src="/images/screenshots/iam/add-arn.png" style="width: 80%" alt="Add ARN">
+</a>
+
+8. Click "Next: Tags". You don't need to add any tags.
+
+9. Click "Next: Review".
+
+10. Enter a suitable name for your Policy.
+
+<a href="/images/screenshots/iam/review-and-create-policy.png" target="_blank">
+  <img src="/images/screenshots/iam/review-and-create-policy.png" style="width: 100%" alt="Review and create policy">
+</a>
+
+11. Click "Create Policy".
+
+The policy giving permissions to invoke a Lambda function has been created. You now need to attach it to the Role that you'll specify in your Ably Lambda Rule.
+
+h3(#attach-policy). Attach the policy to the role
```

Comment:
You can avoid this step if the Policy is created first, and then attached to the Role when it's created.

Reply:
Removed section. I now create a policy and then a role.